packages/contracts/src/Game.sol

```diff
@@ -767,8 +783,11 @@ contract Game {
             revert CardIndexTooHigh();
         }
         uint256 card = gdata.cards[cardIndex];
-        checkPlayProof(pdata, handRoot, card, proof);
+        bytes32 committedSalt = salts[msg.sender];
```

Comment:
The salts hashes should probably be stored in `PlayerData` but let's not change that here, I have it refactored locally, I can merge things.